alonzo/impl/src/Cardano/Ledger/Alonzo/TxWitness.hs

```diff
@@ -244,149 +179,48 @@ instance
   (Era era, ToCBOR (Core.Script era)) =>
   Monoid (TxWitness era)
   where
-  mempty = FlatWitness mempty mempty mempty mempty mempty
+  mempty = TxWitness mempty mempty mempty mempty mempty
 
 -- =======================================================
--- Virtual accessors for the FlatWitness VIEW of TxWitness
+-- Virtual HasField instances for the accessors
 -- =======================================================
 
-witsScript ::
-  (ToCBOR (Core.Script era), Era era) =>
-  TxWitness era ->
-  Map (ScriptHash (Crypto era)) (Core.Script era)
-witsScript (TxWitnessConstr (Memo (TxWitnessRaw _ _ (ScriptData a _ _)) _)) = a
-
-witsData ::
-  (Era era, ToCBOR (Core.Script era)) =>
-  TxWitness era ->
-  Map (DataHash (Crypto era)) (Data era)
-witsData (TxWitnessConstr (Memo (TxWitnessRaw _ _ (ScriptData _ b _)) _)) = b
-
-witsRdmr ::
-  (Era era, ToCBOR (Core.Script era)) =>
-  TxWitness era ->
-  Map RdmrPtr (Data era, ExUnits)
-witsRdmr (TxWitnessConstr (Memo (TxWitnessRaw _ _ (ScriptData _ _ c)) _)) = c
-
--- HasField instances for the virtual accessors
-
 instance
   (Core.Script era ~ script, Crypto era ~ crypto) =>
   HasField "witsScript" (TxWitness era) (Map (ScriptHash crypto) script)
   where
-  getField (TxWitnessConstr (Memo m _)) =
-    _scriptDataScripts
-      . memotype
-      . unScriptDataConstr
-      $ _witsScriptData m
+  getField (TxWitnessConstr (Memo (TxWitnessRaw _ _ s _ _) _)) = s
 
 instance
   (Crypto era ~ crypto) =>
   HasField "witsData" (TxWitness era) (Map (DataHash crypto) (Data era))
   where
-  getField (TxWitnessConstr (Memo m _)) =
-    _scriptDataData
-      . memotype
-      . unScriptDataConstr
-      $ _witsScriptData m
+  getField (TxWitnessConstr (Memo (TxWitnessRaw _ _ _ d _) _)) = d
 
 instance HasField "witsRdmrs" (TxWitness era) (Map RdmrPtr (Data era, ExUnits)) where
```

Comment:
Should be consistent on names, either using `witsFoo` or `txwitsFoo`. I don't really mind which

Reply:
I made it consistent with the specification.